provider/mediaconvert/preset_mapping.go

```diff
@@ -0,0 +1,230 @@
+	container = strings.ToLower(container)
+	switch container {
+	case "m3u8":
+		return mediaconvert.ContainerTypeM3u8, nil
+	case "mp4":
+		return mediaconvert.ContainerTypeMp4, nil
+	default:
+		return "", fmt.Errorf("container %q not supported with mediaconvert", container)
+	}
+}
+
+func h264RateControlModeFrom(rateControl string) (mediaconvert.H264RateControlMode, error) {
+	rateControl = strings.ToLower(rateControl)
+	switch rateControl {
+	case "vbr":
+		return mediaconvert.H264RateControlModeVbr, nil
+	case "", "cbr":
+		return mediaconvert.H264RateControlModeCbr, nil
+	case "qvbr":
+		return mediaconvert.H264RateControlModeQvbr, nil
+	default:
+		return "", fmt.Errorf("rate control mode %q is not supported with mediaconvert", rateControl)
+	}
+}
+
+func h264CodecProfileFrom(profile string) (mediaconvert.H264CodecProfile, error) {
+	profile = strings.ToLower(profile)
+	switch profile {
+	case "baseline":
+		return mediaconvert.H264CodecProfileBaseline, nil
+	case "main":
+		return mediaconvert.H264CodecProfileMain, nil
+	case "", "high":
+		return mediaconvert.H264CodecProfileHigh, nil
+	default:
+		return "", fmt.Errorf("h264 profile %q is not supported with mediaconvert", profile)
+	}
+}
+
+func h264CodecLevelFrom(level string) (mediaconvert.H264CodecLevel, error) {
+	switch level {
+	case "":
+		return mediaconvert.H264CodecLevelAuto, nil
+	case "1", "1.0":
+		return mediaconvert.H264CodecLevelLevel1, nil
+	case "1.1":
+		return mediaconvert.H264CodecLevelLevel11, nil
+	case "1.2":
+		return mediaconvert.H264CodecLevelLevel12, nil
+	case "1.3":
+		return mediaconvert.H264CodecLevelLevel13, nil
+	case "2", "2.0":
+		return mediaconvert.H264CodecLevelLevel2, nil
+	case "2.1":
+		return mediaconvert.H264CodecLevelLevel21, nil
+	case "2.2":
+		return mediaconvert.H264CodecLevelLevel22, nil
+	case "3", "3.0":
+		return mediaconvert.H264CodecLevelLevel3, nil
+	case "3.1":
+		return mediaconvert.H264CodecLevelLevel31, nil
+	case "3.2":
+		return mediaconvert.H264CodecLevelLevel32, nil
+	case "4", "4.0":
+		return mediaconvert.H264CodecLevelLevel4, nil
+	case "4.1":
+		return mediaconvert.H264CodecLevelLevel41, nil
+	case "4.2":
+		return mediaconvert.H264CodecLevelLevel42, nil
+	case "5", "5.0":
+		return mediaconvert.H264CodecLevelLevel5, nil
+	case "5.1":
+		return mediaconvert.H264CodecLevelLevel51, nil
+	case "5.2":
+		return mediaconvert.H264CodecLevelLevel52, nil
+	default:
+		return "", fmt.Errorf("h264 level %q is not supported with mediaconvert", level)
+	}
+}
+
+func h264InterlaceModeFrom(mode string) (mediaconvert.H264InterlaceMode, error) {
+	mode = strings.ToLower(mode)
+	switch mode {
+	case "", "progressive":
+		return mediaconvert.H264InterlaceModeProgressive, nil
+	default:
+		return "", fmt.Errorf("h264 interlace mode %q is not supported with mediaconvert", mode)
+	}
+}
+
+func videoPresetFrom(preset db.Preset) (*mediaconvert.VideoDescription, error) {
```

Comment:
Could we add some tests for this and ``audioPresetFrom``?

Reply:
Do you thing separate tests are really needed for these non-exported methods. This method and `audioPresetFrom ` already have 100% coverage. https://codecov.io/gh/nytimes/video-transcoding-api/pull/223/diff?src=pr&el=tree#D4-123

Reply:
That's a good point, let's leave it as is!